interactive-etl/README.md

```diff
@@ -0,0 +1,330 @@
+User IDs for users who buy 3 or more items over 500 GBP:
+
+```sql
+SELECT 
+    DISTINCT(product_id), 
+    unit_cost, 
+    quantity 
+FROM SalesRecordTable 
+WHERE 
+    quantity >= 3 AND unit_cost > 500;
+```
+
+But there is a problem with this query:
+```shell
+[ERROR] Could not execute SQL statement. Reason:
+java.lang.NumberFormatException: For input string: '£684'. Invalid character found.
+
+```
+
+Who put that `£` sign in there!?! 
+Why is `unit_cost` a string? 
+Indeed, if we check the Avro schema (`data-generator/src/main/resources/sales.avsc`) for the topic we can see: 
+
+```json
+    {"name": "unit_cost",  "type": "string"}
+```
+
+Now, we could go and talk to whoever owns this schema and get them to change it.
+But that would require a redeployment of the producing application and maybe a downstream application expects that currency string and we would be breaking other pipelines if we changed it.
+
+## Cleaning the cost field
+
+All is not lost though, we can work around this formatting issue using Flink SQL.
+
+There are a large number of [built-in functions](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/functions/systemfunctions/) that you can call as part of your queries, including [string manipulation](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/functions/systemfunctions/#string-functions) and [type conversion](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/functions/systemfunctions/#type-conversion-functions).
+We can use those to convert the currency string as part of the query.
+
+In the query below, we add a sub-query that does the conversion of the original values and then select from the output of that:
+
+```sql
+SELECT 
+    DISTINCT(product_id), cost_gbp, quantity 
+FROM (
+    SELECT 
+        product_id, 
+        CAST(TRIM(LEADING '£' from unit_cost) AS INT) AS cost_gbp, 
+        quantity 
+    FROM SalesRecordTable
+) 
+WHERE quantity >= 3 AND cost_gbp > 500;
+```
+
+The query above allows us to treat cost as a proper numeric value, but we are likely to need it in this form for more than just this query.
+If we only need this transformed value for this session we can create a temporary view:
+
+```sql
+CREATE TEMPORARY VIEW CleanedSalesRecords ( 
+    invoice_id, 
+    user_id, 
+    product_id, 
+    quantity, 
+    unit_cost_gbp, 
+    purchase_time
+) AS (
+    SELECT 
+        invoice_id, 
+        user_id,
+        product_id,
+        CAST(quantity AS INT), 
+        CAST(TRIM(LEADING '£' from unit_cost) AS INT), 
+        purchase_time
+    FROM SalesRecordTable
+);
+```
+
+We can then write queries against this view and use the properly formatted cost without having to write the sub-query over again:
+
+```sql
+SELECT 
+    DISTINCT(product_id), 
+    unit_cost_gbp, 
+    quantity 
+FROM CleanedSalesRecords
+WHERE 
+    quantity >= 3 AND unit_cost_gbp > 500;
+```
+
+### Loading back to Kafka
```

Comment:
I know your aiming for a casual/colloquial term but `loading back to Kafka` sounds very strange to me. Maybe `persisting` or `writing to`?

Reply:
I changed it to persisting.